lib/rules/file-extension-in-import.js

```diff
@@ -74,47 +74,56 @@ module.exports = {
             }
 
             // Get extension.
-            const originalExt = path.extname(name)
-            const existingExts = getExistingExtensions(filePath)
-            const ext = path.extname(filePath) || existingExts.join(" or ")
-            const style = overrideStyle[ext] || defaultStyle
+            const currentExt = path.extname(name)
+            const actualExt = path.extname(filePath)
+            const style = overrideStyle[actualExt] || defaultStyle
+
+            const expectedExt = mapTypescriptExtension(
+                context,
+                filePath,
+                actualExt
+            )
 
             // Verify.
-            if (style === "always" && ext !== originalExt) {
-                const fileExtensionToAdd = mapTypescriptExtension(
-                    context,
-                    filePath,
-                    ext
-                )
+            if (style === "always" && currentExt !== expectedExt) {
                 context.report({
                     node,
                     messageId: "requireExt",
-                    data: { ext: fileExtensionToAdd },
+                    data: { ext: expectedExt },
                     fix(fixer) {
-                        if (existingExts.length !== 1) {
-                            return null
-                        }
                         const index = node.range[1] - 1
                         return fixer.insertTextBeforeRange(
                             [index, index],
-                            fileExtensionToAdd
+                            expectedExt
                         )
                     },
                 })
-            } else if (style === "never" && ext === originalExt) {
+            }
+
+            if (
+                style === "never" &&
+                currentExt !== "" &&
+                expectedExt !== "" &&
+                currentExt === expectedExt
+            ) {
```

Comment:
I am incliend to say that we should deprecate the use of `style === 'never'` for a couple of reasons:

1. There are no longer default extensions in esm (https://nodejs.org/api/esm.html#resolution-and-loading-algorithm)
2. Its always more efficient to directly reference the file extension (the resolution algorithm has to work less hard).

I am open to hear what you think about this?